datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedTaskCountEstimator.java

```diff
@@ -27,6 +41,23 @@
    */
   public int getTaskCount(ClusterThroughputInfo throughputInfo, List<String> assignedPartitions,
```

Comment:
One of the major issue I have with task counting using both assigned partition and unassigned partition is, sticky strategy reuses the task object and it will be tricky in this case.

This will not show up as an issue, since we will call the method with assignedPartitions as empty. Do you think, it will make more sense to expose another method `getTaskCount(throughputInfo, unassignedPartitions)` and internally call `getTaskCount(throughputInfo, Collections.emptyList(), unassignedPartitions)`. This way when we later read the code, it will not create unnecessary confusion.

Reply:
I think javadoc and variable naming should take care of that.